File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        maxprofit=0
        min_price=max(prices)
        if len(prices)==0:
            return 0
        for price in prices:
            if price<min_price:
                min_price=price     
            elif price-min_price>maxprofit:
                maxprofit=price-min_price
        return maxprofit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int max_prof=0;
        int buy=prices[0];
        for (int i=1;i<prices.length;i++){
            if (prices[i]<buy){
                buy=prices[i];
            }
            else{
                int profit=prices[i]-buy;
                if (profit>max_prof){
                    max_prof=profit;
                }
            }
        }
        return max_prof;
    }
}
